# <font color='purple'>01-Timescales</font>

## 1. Welcome to your notebooks!

This is a markdown cell. 

It is a very flexible way to include explanations of the code cell that follows.

It supports HTML type formatting. For example, here's a <font color='red'> sentence written in red </font>.

It can also interpret latex commands for math symbols:

$$ i \hbar \frac{\partial}{\partial t} \Psi(\mathbf{r}, t) = \left[ -\frac{\hbar^2}{2\mu}\nabla^2 + V(\mathbf{r}, t) \right] \Psi(\mathbf{r}, t).$$

Using one \$ creates a inline equation or symbol e.g. $\alpha=\pi$.

See a list of useful math symbols [here](https://www.math.uci.edu/~xiangwen/pdf/LaTeX-Math-Symbols.pdf)

#### We start by importing the modules
* Numpy -- operations on arrays and matrixes (and pi)
* Astropy constants -- defines physical and astronomical constants
* Astropy units -- defined quantities with units. We also import the CDS conversions


> **Note:**<font color='blue'> To execute the code in a cell, use "Shift - enter"</font>

In [1]:
from astropy import constants as const
import astropy.units as u
from astropy.units import cds
cds.enable() 

import numpy as np

Here's a few constants that we will need later

In [2]:
Lsun = const.L_sun.decompose()
Msun = const.M_sun.decompose()
Rsun = const.R_sun.decompose()
G = const.G.decompose()
c = const.c.decompose()

## 2. Kelvin-Helmholtz timescale

We want to find the potential energy with 
$$ \Delta \Omega = \frac{3}{5} \frac{G M_\odot^2}{R_\odot}$$

Accordign to the virial theorem, half of the potential energy lost goes into termal energy, and the other half escapes into radiation. The Kelvin-Helmholtz timescale will therefore be:
$$ \frac{\Delta \Omega}{2 L_\odot}$$

In [26]:
##################
## In class

PotentialEnergy = 3/5*G*Msun**2/Rsun
print(f"The energy budget form gravitation contraction is: {PotentialEnergy.to(u.J)}")

timescale = 1/2 * PotentialEnergy / Lsun
print(f"The timescale is: {timescale.to(u.Myr)}")

The energy budget form gravitation contraction is: 2.2758665429963157e+41 J
The timescale is: 9.419783028667418 Myr


> **TODO**: <font color='red'>Please write a small paragraph with an interpretation of the result obtained:</font>

<font color='green'> **Sid**: </font> Not long enough to explain carbon dating

## 3. Nuclear timescale

> **TODO** <font color='red'>: Write a brief paragraph that describe the calculation we make below. Use Latex formatting to render the math</font>

In [29]:
##################
## In class

EnergyBudget = 0.1*0.007*Msun*c**2

# Only the core participates so 0.1

print(f"Energy budget of the Sun: {EnergyBudget.to(u.J)}")

newtimescale = EnergyBudget / Lsun
print(f"The timescale is: {newtimescale.to(u.Gyr)}")

Energy budget of the Sun: 1.2509655681188957e+44 J
The timescale is: 10.35546153993683 Gyr


> **TODO**: <font color='red'>Please write a small paragraph with an interpretation of the result obtained:</font>


## 4. Hydrodynamic timescale
> **TODO** <font color='red'>: Write a brief paragraph that describe the calculation we make below. Use Latex formatting to render the math</font>

Kinematics:

$ \Delta t \sim \bigg(\dfrac{R_{*}}{g}\bigg)^{1/2}$

where $g \sim \dfrac{G M}{R^2} $

$\implies \Delta t \sim \bigg(\dfrac{R_{*}^3}{GM}\bigg)^{1/2}$

In [34]:
##################
## In class

Timescale = (Rsun**3 / (G*Msun))**(0.5)
print(f"The timescale is: {Timescale.to(u.min)}")

The timescale is: 26.54764109394458 min


So, it would take 26.5 mins for sun to collapse if there were no hydrostatic pressure.

We consider stars as quasistatic.

> **TODO**: <font color='red'>Please write a small paragraph with an interpretation of the result obtained:</font>


## **5. Reading assignement: "what can we measure about stars"**

In this graduate course, we are making an advanced physical and mathematical model of star. 

But it is still good to learn and/or remind ourselves about which physical characteristics of stars we can actually measure. You might have covered some of this in some details in previous physics or astro courses (at UD PHYS 133, 144, 333, or 469) -- but it is still a good idea to have a quick look at the suggested reading below before crafting your paragraph. 

> **TODO**: <font color='red'>Have a look at section 5.2 of the Open Stack Astronomy online [textbook](https://openstax.org/books/astronomy/pages/5-2-the-electromagnetic-spectrum), and write a short conceptual paragraph about how the law of radiation allows astronomer to determine the temperature of stars. </font>